SonarQube.Common/AnalysisConfig/AnalysisConfigExtensions.cs

```diff
@@ -43,80 +44,113 @@ public static string GetSetting(this AnalysisConfig config, string settingId, st
         }
 
         /// <summary>
-        /// Attempts to find and return the analysis setting with the specified id
+        /// Sets the value of the additional setting. The setting will be added if it does not already exist.
         /// </summary>
-        /// <returns>True if the setting was found, otherwise false</returns>
-        public static bool TryGetSetting(this AnalysisConfig config, string settingId, out AnalysisSetting result)
+        public static void SetConfigValue(this AnalysisConfig config, string settingId, string value)
+        {
+            SetValue(config, settingId, value);
+        }
+
+        /// <summary>
+        /// Returns a provider containing all of the analysis settings from the config.
+        /// Optionally includes settings downloaded from the SonarQube server.
+        /// </summary>
+        public static IAnalysisPropertyProvider GetAnalysisSettings(this AnalysisConfig config, bool includeServerSettings)
```

Comment:
The MSBuild task needs to take include the settings downloaded from the SonarQube server. However, the server settings should be ignored when generating the sonar-project.properties file.